experimental/overhead_matching/kimera_spectacular_data_provider_test.cc

```diff
@@ -0,0 +1,178 @@
+
+#include "experimental/overhead_matching/kimera_spectacular_data_provider.hh"
+
+#include <iostream>
+#include <sstream>
+
+#include "common/video/image_compare.hh"
+#include "experimental/overhead_matching/spectacular_log.hh"
+#include "fmt/format.h"
+#include "gtest/gtest.h"
+#include "kimera-vio/pipeline/Pipeline-definitions.h"
+#include "opencv2/opencv.hpp"
+
+namespace robot::experimental::overhead_matching {
+
+std::ostream& operator<<(std::ostream& out, const time::RobotTimestamp& t) {
+    std::ostringstream ss;
+    ss << std::fixed << std::setprecision(9)
+       << std::chrono::duration<double>(t.time_since_epoch()).count();
+    out << ss.str();
+    return out;
+}
+
+bool compare_imu_samples(const robot::experimental::overhead_matching::ImuSample& robot_imu,
+                         const VIO::ImuMeasurement& kimera_imu) {
+    // timestamp
+    if (kimera_imu.timestamp_ != robot_imu.time_of_validity.time_since_epoch().count()) {
+        std::cout << "kimera timestamp " << kimera_imu.timestamp_ << std::endl;
+        std::cout << "robot timestamp " << robot_imu.time_of_validity.time_since_epoch().count()
+                  << std::endl;
+        std::cout << "diff "
+                  << robot_imu.time_of_validity.time_since_epoch().count() - kimera_imu.timestamp_
+                  << std::endl;
+        std::cout << "timestamp" << std::endl;
```

Comment:
No need to change, but I've really been liking using fmtlib. It allows you do to python format string style interpolation. This would become:
```
fmt::println("Kimera timestamp: {}", kimera_imu_timestamp_);
fmt::println("robot timestamp: {}", robot_imu.time_of_validity);
...
```